### PR TITLE
Fix stylelint source order/specificity offenders

### PR DESCRIPTION
**Pull request by thefifthisa**
Partial fix for #1119

## Description:
Added `no-descending-specificity` rule to stylelint config and rearranged CSS selectors in 16 files to have selectors with lower specificity come first.

### CODE REVIEW

File: static/css/components/home.less [hunk 1]
```diff
@@ -70,6 +69,11 @@ div.chartHome a:hover {
   }
 }
 
+#home-stats-charts .statschart:hover a {
```
Comment (jdlrobson) [inline static/css/components/home.less]: This adds some additional CSS

I think in this file it makes sense to wrap the rule with 
```
/* stylelint-disable no-descending-specificity */ 
a:hover...
/* stylelint-enable no-descending-specificity */ 
```
and leave this file unchanged.

File: static/css/components/ui-tabs.less [hunk 2]
```diff
@@ -43,6 +43,11 @@
         text-decoration: underline;
       }
     }
+    a:hover,
```
Comment (jdlrobson) [inline static/css/components/ui-tabs.less]: This adds some additional CSS

I think in this file it makes sense to wrap the rule with 
```
/* stylelint-disable no-descending-specificity */ 
a:hover...
/* stylelint-enable no-descending-specificity */ 
```
and leave this file unchanged.

File: static/css/components/wmd.less [hunk 3]
```diff
@@ -10,11 +10,14 @@
   background-color: transparent;
 }
 
+fieldset.major .wmd-button-bar {
```
Comment (jdlrobson) [inline static/css/components/wmd.less]: This adds some additional CSS

I think in this file it makes sense to wrap the rule with 
```
/* stylelint-disable no-descending-specificity */ 
a:hover...
/* stylelint-enable no-descending-specificity */ 
```
and leave this file unchanged.

**Comment by jdlrobson** [comment]
Thank you @thefifthisa for taking this on!
Sorry about my nitpicks! I hope they make sense!

**Comment by mekarpeles** [comment]
@thefifthisa I think we're very close -- would you like help resolving merge conflicts and responding to the code review suggestions?

**Comment by jdlrobson** [comment]
To rebase you'll need to pull down the latest version of openlibrary and then rebase your changes on top of them like so:
```
git remote add upstream https://github.com/internetarchive/openlibrary.git
git fetch upstream
git rebase upstream/master
```
When I do this, it seems to be able to do it all automatically, which is a relief and should make this easier :) It's always a pain when there are manual corrections that need to be made in this process.

You can then force push the new version using
```
git push origin 1119/refactor/fix-stylelint-source-order-offenders --force
```

**Comment by thefifthisa** [comment]
@jdlrobson done!

**Comment by jdlrobson** [comment]
You nailed it!
awesome!!

**Pull request merged**